src/jit/lower.cpp

```diff
@@ -3118,13 +3118,17 @@ GenTree* Lowering::LowerVirtualStubCall(GenTreeCall* call)
 
         // Direct stub calls, though the stubAddr itself may still need to be
         // accesed via an indirection.        
-        GenTree* tmp = AddrGen(stubAddr);
-        tmp->gtRegNum = REG_VIRTUAL_STUB_PARAM;
-
-        tmp = Ind(tmp);
-        tmp->gtRegNum = REG_JUMP_THUNK_PARAM;
-
-        result = tmp;
+        GenTree* addr = AddrGen(stubAddr);
+        GenTree* indir = Ind(addr);
```

Comment:
Regarding the first comment, I think it would be complex to add asserts in codegen for the various expectations regarding calls. If I added a check that the virtual stub calls have one of the expected forms, I would imagine that we should probably add checks for many other types, which would essentially duplicate the complexity of the code in lower. The only thing that might be reasonable would be to have a method that produces a characterization of a call, which lower could use to set up the control expression, and which codegen could use to validate it. But that would be a non-trivial restructuring, I think.